code/game/mecha/working/ripley.dm

```diff
@@ -31,8 +31,9 @@
 					ore.forceMove(ore_box)
 
 /obj/mecha/working/ripley/Destroy()
-	for(var/i=1, i <= hides, i++)
-		new /obj/item/stack/sheet/animalhide/goliath_hide(loc) //If a goliath-plated ripley gets killed, all the plates drop
+	GET_COMPONENT(C,/datum/component/armor_plate)
+	for(var/i=1, i <= C.amount, i++)

```

Comment:
`for(var/i in 1 to C.amount)`

Reply:
also not sure how I feel about this, the concept of dropping the upgrades on destruction of parent should be on the component as an optional.